### ISSUE TITLE
Filter on project id in typeaheads

### ISSUE THREAD

**Issue by KasperBrandt** [akvo/akvo-rsr#1751]
In addition to the project name, it should also be possible to filter on the project id in the typeaheads of the project editor.

**Comment by gabemart** [comment]
Would it be acceptable for the ID to appear in brackets after the project name? E.g. `My Test Project (61)` or possibly `My Test Project (ID: 61)`

**Comment by KasperBrandt** [comment]
Yup, definitely (right @lauraroverts?). I would personally go for the latter ```My Test Project (ID: 61)```

**Comment by gabemart** [comment]
Implemented now as `My Test Project (ID: 61)` but this is trivial to change if needed

**Comment by gabemart** [comment]
# Test Plan

1) Open the project editor to section 1

2) Hit "Add another related project" and select the "Related RSR project" input

3) Verify typing a project's ID filters to that project in the list of options

4) Verify typing a project's name filter to that project

5) Verify projects in the list of options display in the `ProjectName (ID: projectID)` format

6) Verify saving works properly

**Comment by gabemart** [comment]
I'm merging issue #1752 into this issue as the majority of the code is the same.

>In all organisation typeaheads, it should be possible to filter on (and show?) the long and short names.

**Comment by gabemart** [comment]
I've made changes so you can *filter* by anything in the long name or short name or an org, but only the short name is displayed as an option.

For example, on my local instance, typing "foundation" gives the options "Akvo" (for "Akvo Foundation") and "Lagom" (for "The Lagom Foundation"). 

![screen shot 2015-09-16 at 16 13 20](https://cloud.githubusercontent.com/assets/2589977/9908965/fc8ae382-5c8d-11e5-86d3-f99a4944f5e4.png)

It might be confusing to filter for text that isn't displayed to the user, but it would be very inelegant to display both the short name and long name as options.

Perhaps we should simply always display the long name, but allow filtering by both long name and short name?

@KasperBrandt @lauraroverts - thoughts?

**Comment by KasperBrandt** [comment]
@gabemart The project typeahead looks good, although I noticed that we didn't show non-published projects, so I've added those (be sure to pull the latest version of the branch).

On the organisations; I think it's more obvious if we show ```long name (short name)```. Although sometimes the long name is the same as the short name, in that case we should show ```long name```. And sometimes there is not long name, in that case we should show ```short name```. We should be able to add this logic here I guess: https://github.com/akvo/akvo-rsr/pull/1811/files#diff-7bd4789921072174d75e2fc028733572R850

Also, I noticed that the label of participating organisations is now 'reporting organisation'. Not sure where that stems from, but that should be fixed too. Note; I typed ```Akvo Foundation (Akvo)``` in the image myself to see how it looks:
<img width="950" alt="screen shot 2015-09-17 at 10 37 13" src="https://cloud.githubusercontent.com/assets/1497367/9927710/81c0e09c-5d29-11e5-8d09-e78bce46336b.png">
